### PR TITLE
Test Windows via GH actions

### PR DESCRIPTION
**Pull request by jarrodmillman**
@rossbar Switch to GH actions from Appveyor for Windows testing.  Once this is merged, I will clean up #4561 so that it is just focused on pygraphviz testing on Windows.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -21,17 +21,20 @@ jobs:
...
+          python -m pip install --upgrade pip wheel setuptools
+          python -m pip install -r requirements.txt
+          python -m pip install .
+          python -m pip list
+        else
+          pip install --upgrade pip wheel setuptools
+          pip install -r requirements.txt
+          pip install .
+          pip list
+        fi
```
Comment (rossbar) [inline .github/workflows/test.yml:37]:
```suggestion
        python -m pip install --upgrade pip wheel setuptools
        python -m pip install -r requirements.txt
        python -m pip install .
        python -m pip list
```
I think you can get away with the pip module invocation on all platforms which would get rid of the conditional (unless you tried this on mac/linux already and something went wrong). Not critical though.

**Review by rossbar — APPROVED** [review]
LGTM - I assume the failing appveyor job will stop showing up once this is merged, or maybe there's some additional switch-flipping that needs to be done in the appveyor account. Either way, +1 for switching the windows testing to gh workflows.

File: .github/workflows/test.yml (hunk 1 above)
Comment (jarrodmillman) [reply .github/workflows/test.yml:37]: Windows won't let you update pip when it is running.

Comment (jarrodmillman) [reply .github/workflows/test.yml:37]: Never mind, I misread what you wrote.

**Comment by stefanv** [comment]
This is a very satisfying cleanup!  It would have been nice if we could have defined the `pip` command per platform, but that seems oddly hard to do on GHA.

**Pull request merged**